misk/src/main/kotlin/misk/web/extractors/StringConverter.kt

```diff
@@ -0,0 +1,58 @@
+package misk.web.extractors
+
+import kotlin.reflect.KType
+import kotlin.reflect.full.createType
+import kotlin.reflect.jvm.javaType
+
+// Pre-generate these specific handled KTypes so we don't need to regenerate them for every
+// query parameter for every call.
+val stringType: KType = String::class.createType(nullable = false)
+val stringTypeNullable: KType = String::class.createType(nullable = true)
+val intType: KType = Int::class.createType(nullable = false)
+val intTypeNullable: KType = Int::class.createType(nullable = true)
+val longType: KType = Long::class.createType(nullable = false)
+val longTypeNullable: KType = Long::class.createType(nullable = true)
+val doubleType: KType = Double::class.createType(nullable = false)
+val doubleTypeNullable: KType = Double::class.createType(nullable = true)
+val booleanType: KType = Boolean::class.createType(nullable = false)
+val booleanTypeNullable: KType = Boolean::class.createType(nullable = true)
+
+internal typealias StringConverter = (String) -> Any?
+
+internal fun converterFor(type: KType): StringConverter? = when (type) {
+  stringType -> { it -> it }
+  stringTypeNullable -> { it -> it }
+  intType -> numberConversionWrapper { it.toInt() }
+  intTypeNullable -> numberConversionWrapper { it.toInt() }
+  longType -> numberConversionWrapper { it.toLong() }
+  longTypeNullable -> numberConversionWrapper { it.toLong() }
+  doubleType -> numberConversionWrapper { it.toDouble() }
+  doubleTypeNullable -> numberConversionWrapper { it.toDouble() }
+  booleanType -> numberConversionWrapper { it.toBoolean() }
+  booleanTypeNullable -> numberConversionWrapper { it.toBoolean() }
+  else -> createFromValueOf(type)
+}
+
+fun numberConversionWrapper(wrappedFunc: StringConverter): StringConverter {
+  return {
+      try {
+        wrappedFunc(it)
+      } catch(e: NumberFormatException) {
+        throw IllegalArgumentException(
```

Comment:
nit: don't wrap unless needed

Reply:
I was just trying to avoid repeating the NumberFormatException catch a bunch of times. Is there a prettier way to do that in Kotlin?

Reply:
Sorry I mean don't line wrap :)